### ISSUE TITLE
MongoDB based device registry

### ISSUE THREAD

**Issue by mhemmeter** [eclipse/hono#1679]
As of today Eclipse Hono provides file based implementations of the following APIs: Device Registry Management API, Device Registration API, Credentials API and Tenant API. These implementations are not meant for productive use. Therefore it is required that adopters implement these APIs or use an existing system in their concrete deployment scenario. The latter one is however not always existing.

To simplify usage of Eclipse Hono and increase adoption I think it makes sense to provide a production ready implementation of these APIs. 

I see the following requirements for a production ready implementation:
- The implementation is built on top of a scalable persistence layer. In our deployment of Eclipse Hono a MongoDB is used and it scales very well. Also Eclipse Ditto provides a persistence layer built on top of MongoDB, so using that technology simplifies the combined usage of different Eclipse IoT projects.
- All implementations are secured with an authentication mechanism

**Comment by sophokles73** [comment]
I sympathize with the idea of having such a Mongo DB based implementation :+1:

**Comment by mhemmeter** [comment]
I would propose to reduce the scope of this issue and focus on building the implementation on top of MongoDB as the persistence layer. We can secure the API in a follow-up issue I will create.

**Comment by sophokles73** [comment]
@kaniyan anything left to do for this issue?

**Comment by kaniyan** [comment]
@sophokles73 The implementation is over. The documentation needs to be done.

**Comment by sophokles73** [comment]
@kaniyan ok, are you already working on that?

**Comment by kaniyan** [comment]
@sophokles73 Currently I am working on helm charts for this mongo db based device registry, thereby to make a PR in the IoT packages repo. After that I can start with the documentation. We could also close this issue and I can open a new one to track the documentation separately.

**Comment by mhemmeter** [comment]
From my perspective documentation should be part of this issue.

**Comment by sophokles73** [comment]
@kaniyan Can you please first add the documentation for the Mongo DB registry and only after that is finished continue with the PR for the Hono Helm chart?

**Comment by kaniyan** [comment]
@sophokles73 sure, I can proceed with the documentation first and thereby we can close this issue and then proceed with the Helm chart.

**Comment by kaniyan** [comment]
@mhemmeter @sophokles73 The documentation for the _MongoDB based device registry_ is now over with the merge of PRs #2035 and #2027. We can now close this issue.

**Comment by mhemmeter** [comment]
Thanks, @kaniyan. I'm closing the issue.
